client/src/request.js

```diff
@@ -1,38 +1,63 @@
-const photos = (productId) => {
-  return fetch(`http://ec2-3-136-203-39.us-east-2.compute.amazonaws.com:4002/photos/id/${productId}`)
-  .then(res => res.json())
-  .then(productPhotos => {
-    return productPhotos;
-  })
-  .catch(error => {
+const fetchWithTimeout = async (resource, options) => {
+  const { timeout = 4000 } = options;
+
+  const controller = new AbortController();
+  const id = setTimeout(() => controller.abort(), timeout);
+
+  const response = await fetch(resource, {
+    ...options,
+    signal: controller.signal
+  });
+  clearTimeout(id);
+
+  return response;
+};
+
+const photos = async (productId) => {
+
+  try {
+    const response = await fetchWithTimeout(`http://ec2-3-136-203-39.us-east-2.compute.amazonaws.com:4002/photos/id/${productId}`, {
+      timeout: 3000
+    });
+    const productPhotosInfo = await response.json();
+    return productPhotosInfo;
+  } catch (error) {
     let productPhotos = ['https://via.placeholder.com/640x480?text=Unable+to+get+primary+product+image'];
     for (let i = 2; i <= 5; i++) {
       productPhotos.push(`https://via.placeholder.com/640x480?text=Unable+to+get+product+image+${i}`);
     };
-    return productPhotos;
-  });
+    let errorProductPhotos = {
+      primaryUrl: productPhotos[0],
+      productUrls: productPhotos,
+    }
+    return errorProductPhotos;
+  }
+
 };
 
-const productInfo = (productId) => {
-  return fetch((`http://ec2-18-217-85-161.us-east-2.compute.amazonaws.com:4004/description/${productId}`))
-  .then(res => res.json())
-  .then((response) => {
-    let rawItemInfo = response[0];
+const productInfo = async (productId) => {
+
+  try {
+    const response = await fetchWithTimeout(`http://ec2-18-217-85-161.us-east-2.compute.amazonaws.com:4004/description/${productId}`, {
+      timeout: 3000
```

Comment:
Any particular reason to have the code timeout after 3 seconds?

Reply:
Well, the default was originally 8 seconds but I felt like that was too long to have nothing displaying on the screen. So if the network request doesn't happen in 3 seconds then you get the unable to display product images placeholders instead of the product photos.